frontend/test/admin/datamodel/FieldApp.integ.spec.js

```diff
@@ -307,12 +308,14 @@ describe("FieldApp", () => {
         .first();
       click(useFKButton);
       store.waitForActions([UPDATE_FIELD_DIMENSION, FETCH_TABLE_METADATA]);
-      // TODO: Figure out a way to avoid using delay – the use of delays may lead to occasional CI failures
-      await delay(500);
 
-      const fkFieldSelect = section.find(SelectButton);
+      let fkFieldSelect;
+
+      await eventually(() => {
+        fkFieldSelect = section.find(SelectButton);
```

Comment:
FYI I'd just do `expect(section.find(SelectButton).text()).toBe("Name");` not a big deal if you want to merge as is